testing/BUILD.gn

```diff
@@ -17,13 +17,16 @@ source_set("testing_lib") {
 
   sources = [
     "assertions.h",
+    "post_task_sync.cc",
+    "post_task_sync.h",
```

Comment:
I am not sure if this is a good place to add `post_task_sync.h` and `post_task_sync.cc`. In addition, I am not sure whether `post_task_sync.h` and `post_task_sync.cc` are good file names.
But I think we should put these helper functions in a public place like this so that other unit tests can use them directly

Reply:
Yea, this sounds good to me.